### ISSUE TITLE
No gutters between columns when updating dash and dash-bootstrap-components 

### ISSUE THREAD

**Issue by lhsusan1016** [facultyai/dash-bootstrap-components#448]
- dash version: 1.16.0
- dash-bootstrap-components version: 0.10.5
- components affected by bug:

### What is happening?

We updated dash and dash-bootstrap-components to the lastest version. All the columns do not show no gutters as before any more but when setting the overflowX to auto, the gutters come back. 
If we set no_gutters=True, the two columns overlap. If we set form=True, it seems the same as no_gutters=True.

### What should be happening?

There are gutters between columns without any optional arguments in the previous versions.

**Comment by tcbegley** [comment]
Hi @lhsusan1016 

Thanks for reporting, could you share some example code? I was unable to reproduce with the following code

```python
import dash
import dash_bootstrap_components as dbc

app = dash.Dash(external_stylesheets=[dbc.themes.BOOTSTRAP])

app.layout = dbc.Container(
    [
        dbc.Row(
            [dbc.Col(dbc.Card(f"Card {i + 1}", body=True)) for i in range(3)]
        )
    ],
    className="mt-5",
)

if __name__ == "__main__":
    app.run_server(debug=True)
```

Which gives me this

![image](https://user-images.githubusercontent.com/15220906/92355053-661e7e80-f0db-11ea-9f33-cec1d1cbfb3f.png)

running

```
dash                            1.16.0
dash-bootstrap-components       0.10.5
```

**Comment by lhsusan1016** [comment]
Thanks for your reply!

Code:
`import dash
import dash_table as dt
import dash_bootstrap_components as dbc

app = dash.Dash(external_stylesheets=[dbc.themes.BOOTSTRAP])

app.layout = dbc.Container([
    dbc.Row([
        dbc.Col(dt.DataTable(
            id='data_table',
            columns=[{'id': 'Name', 'name': 'Name'}],
            data=[{'Name': ''}],
            editable=True,
            # style_table={'overflowX': 'auto'},
        ),
            width=4
        ),
        dbc.Col(dt.DataTable(
            id='data_table2',
            columns=[{'id': 'Name', 'name': 'Name'}],
            data=[{'Name': ''}],
            editable=True,
            # style_table={'overflowX': 'auto'},
        ),
            width=4
        )]
    )],
    className="mt-5",
)

if __name__ == "__main__":
    app.run_server(debug=True)`

which gives me:
![image](https://user-images.githubusercontent.com/10786183/92356822-558cf900-f119-11ea-835e-ad93566b3555.png)

with style_table={'overflowX': 'auto'},

I get:
![image](https://user-images.githubusercontent.com/10786183/92356897-76ede500-f119-11ea-8ddf-39a973ebf56b.png)

**Comment by tcbegley** [comment]
The problem is probably because `dash-table` sets the `row` class on rows of the table, and Bootstrap also uses a class called `row` for its rows. In particular Bootstrap rows set negative margins (so that columns at the end of the row are flush with the content) which is why your gutters are disappearing.

You can override this either using the `css` argument to `DashTable`

```python
dt.DataTable(..., css=[{"selector": ".row", "rule": "margin: 0"}])
```

or you can put a snippet like the following in your `assets/` folder

```css
.dash-table-container .row {
  display: block;
  margin: 0;
}
```

I don't know why this would have changed between versions of Dash / dash-bootstrap-components though...

**Comment by lhsusan1016** [comment]
Thanks a lot!!!

I have tried to use the css to DashTable and it works when not using overflowX:
![image](https://user-images.githubusercontent.com/10786183/92366207-6cd2e300-f127-11ea-995d-e13c48debe57.png)

But if I add style_table={'overflowX': 'auto'}, it doen't work well as follows:
![image](https://user-images.githubusercontent.com/10786183/92366134-5462c880-f127-11ea-89ec-1926280d680a.png)

The content in the data table may be long so we want to use overflowX and do you know how to fix this?

And you have mentioned to put a snippet, could you explain more where to put it?

Thank you very much for your help!

**Comment by tcbegley** [comment]
> And you have mentioned to put a snippet, could you explain more where to put it?

See [here](https://dash.plotly.com/external-resources) for how to add custom CSS. Easiest thing is to create a folder called `assets/` and save the CSS in there, Dash will automatically discover it and link it.

Regarding the margin, I took a look and it seems that dash-table is setting `margin-left: -1px` for some reason, which is why the border is disappearing. I'm not so familiar with dash-table, but if you figure out how to override that you'll fix the problem.

![image](https://user-images.githubusercontent.com/15220906/92567435-9b5ed400-f275-11ea-8b33-cdf400da8c2f.png)

**Comment by tcbegley** [comment]
Issues with dash-table should have been resolved by the latest version of Dash (1.18.1+). I've also noted the issue in the new [FAQ section](https://dash-bootstrap-components.opensource.faculty.ai/docs/faq/) in the docs for those stuck with historical versions of Dash.
